src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -464,34 +469,153 @@ std::vector< T > collectUniqueValues( std::vector< T > const & data )
 
   return allData;
 }
+/**
+ * @brief Identify the type of the polyhedron
+ *
+ * @param cell The VTK cell, type polyhedron (42)
+ * @return int The type of polyhedron:
+ *              =0 represent a general polyhedron
+ *              >0 correspond to the equivalent standard vtk cell (10, 12, 13, 14, 15, 16)
+ *             N<0 identify a prism with (-N) sides
+ */
+int polyhedronType( vtkCell * const cell )
+{
+  localIndex numberOfFaces = cell->GetNumberOfFaces();
+  localIndex numberOfQuads = 0;
+  localIndex numberOfTriangles = 0;
+  set< localIndex > ListQuadsPoints;
+  set< localIndex > ListNoQuadsPoints;
```

Comment:
`std::set` is a really expensive container because it makes a separate memory allocation for each inserted element, and we are calling this in a big loop over many cells. Luckily, we have a flatset implementation in `LvArray`, and it can be backed by a stack-allocated buffer, resulting in zero dynamic allocations. The typing of it is somewhat involved: first, outside the function, we have to declare:
```c++
template< typename T, int MAX_SIZE >
struct StackSetHelper
{
  template< typename U >
  using Buffer = LvArray::StackBuffer< U, MAX_SIZE >;

  using type = LvArray::SortedArray< T, localIndex, Buffer >;
};

template< typename T, int MAX_SIZE >
using StackSet = typename StackSetHelper< T, MAX_SIZE >::type;
```
and then just use it like this:
```suggestion
  StackSet< localIndex, 64 > ListQuadsPoints;
  StackSet< localIndex, 64 > ListNoQuadsPoints;
```
where 64 is the maximum number of unique points we expect to see in any polyhedral element - you can make it as large as you want (stack allocation is almost free), but realistically I would use a value in 64-128 range. You won't be able to compare these using `==`, so instead use
```c++
std::equal( ListQuadsPoints.begin(), ListQuadsPoints.end(),
            ListNoQuadsPoints.begin(), ListNoQuadsPoints.end() )
```